src/frontend/gateways/Api.gateway.ts

```diff
@@ -82,11 +83,18 @@ const ApiGateway = () => ({
     });
   },
   listAds(contextKeys: string[]) {
-    return request<Ad[]>({
-      url: `${basePath}/data`,
-      queryParams: {
-        contextKeys,
-      },
+    // TODO: Figure out a better way to do this so session ID gets propagated to
+    // all endpoints
+    const baggage = propagation.getActiveBaggage() || propagation.createBaggage();
+    const newBaggage = baggage.setEntry('app.session.id', { value: userId });
```

Comment:
We should use the AttributeNames enum for the baggage key.

```
import {AttributeNames} from "../utils/enums/AttributeNames";
```

then use `AttributeNames.SESSION_ID`.